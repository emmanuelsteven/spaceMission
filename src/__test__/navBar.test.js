import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import NavBar from '../component/navBar';

const mockStore = configureStore([]);

describe('NavBar', () => {
  let store;

  beforeEach(() => {
    store = mockStore({});
  });

  it('renders correctly', () => {
    render(
      <Provider store={store}>
        <Router>
          <NavBar />
        </Router>
      </Provider>,
    );

    expect(screen.getByAltText('navlogo')).toBeInTheDocument();
    expect(screen.getByText('Space Travelers Hub')).toBeInTheDocument();
    expect(screen.getByText('Rockets')).toBeInTheDocument();
    expect(screen.getByText('Missions')).toBeInTheDocument();
    expect(screen.getByText('My Profile')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <NavBar />
        </Router>
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
