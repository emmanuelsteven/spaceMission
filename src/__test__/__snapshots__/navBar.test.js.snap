// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="nav-container"
  >
    <div
      class="nav-logo"
    >
      <img
        alt="navlogo"
        height="50"
        src="planet.png"
        width="50"
      />
      <h1
        class="navtitle"
      >
        <a
          activeclassname="active"
          aria-current="page"
          class="active"
          href="/"
        >
          Space Travelers Hub
        </a>
      </h1>
    </div>
    <div
      class="navlinks"
    >
      <a
        activeclassname="active"
        aria-current="page"
        class="navlink active"
        href="/"
      >
        Rockets
      </a>
      <a
        activeclassname="active"
        class="navlink"
        href="/mission"
      >
        Missions
      </a>
      <a
        activeclassname="active"
        class="navlink profile"
        href="/My-Profile"
      >
        My Profile
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
